examples/plot_meg_example.py

```diff
@@ -0,0 +1,184 @@
+        forward, evoked.info, noise_cov, pca=False, depth=depth,
+        loose=loose, weights=None, weights_min=None, rank=None)
+
+    # Select channels of interest
+    sel = [all_ch_names.index(name) for name in gain_info['ch_names']]
+    M = evoked.data[sel]
+
+    # Whiten data
+    M = np.dot(whitener, M)
+
+    n_orient = 1 if is_fixed_orient(forward) else 3
+
+    X, active_set = solver(M, gain, n_orient)
+    X = _reapply_source_weighting(X, source_weighting, active_set)
+
+    stc = _make_sparse_stc(X, active_set, forward, tmin=evoked.times[0],
+                           tstep=1. / evoked.info['sfreq'])
+
+    return stc
+
+
+###############################################################################
+# Define your solver
+
+def solver(y_train, X_train, n_orient):
+    """Run L2 penalized regression and keep 10 strongest locations.
+
+    Parameters
+    ----------
+    M : array, shape (n_channels, n_times)
+        The whitened data.
+    G : array, shape (n_channels, n_dipoles)
+        The gain matrix a.k.a. the forward operator. The number of locations
+        is n_dipoles / n_orient. n_orient will be 1 for a fixed orientation
+        constraint or 3 when using a free orientation model.
+    n_orient : int
+        Can be 1 or 3 depending if one works with fixed or free orientations.
+        If n_orient is 3, then ``G[:, 2::3]`` corresponds to the dipoles that
+        are normal to the cortex.
+
+    Returns
+    -------
+    X : array, (n_active_dipoles, n_times)
+        The time series of the dipoles in the active set.
+    active_set : array (n_dipoles)
+        Array of bool. Entry j is True if dipole j is in the active set.
+        We have ``X_full[active_set] == X`` where X_full is the full X matrix
+        such that ``M = G X_full``.
+    """
+    n_times = y_train.shape[1]
+    idx_max = np.argmax(np.sum(y_train ** 2, axis=0))
+    y_train = y_train[:, idx_max]
+
+    n_samples, n_features = X_train.shape
+    alpha_max = (np.abs(X_train.T @ y_train)).max() / n_samples
+    X_train /= alpha_max
+
+    alpha_max = (np.abs(X_train.T @ y_train)).max() / n_samples
+    alpha0 = 0.7 * alpha_max
+    log_alpha0 = np.log(alpha0)
+
+    tol = 1e-5
+    model = "wlasso"
+    criterion = "sure"
+    n_outer = 10
+
+    if model == "wlasso":
+        log_alpha0 = log_alpha0 * np.ones(n_features)
+
+    model = wLasso(X_train, y_train, log_alpha0)
+    sigma = 1 / np.sqrt(50)
+    criterion = SURE(X_train, y_train, model, sigma)
+    algo = ImplicitForward(criterion)
+    monitor_grad = Monitor()
+    grad_search(
+        algo, log_alpha0, monitor_grad, n_outer=n_outer, tol=tol)
+
+    X = criterion.dense0[:, np.newaxis] * np.ones((1, n_times))
+    active_set = criterion.mask0
+    X /= alpha_max
+
+    return X, active_set
+
+
+loose, depth = 0., .8  # corresponds to free orientation
+stc = apply_solver(solver, evoked, forward, noise_cov, loose, depth)
```

Comment:
it would be great to also return the residual to get an idea of the goodness of fit. Did you underfit a lot or non?